### PR TITLE
BUG:  incorrect warning in read_file for pyogrio when layer keyword is provided

### PR DESCRIPTION
**Pull request by FredericF**
Bug is described in issue #3377

Solution is to pass down the layer argument from gpd.read_file to pyogrio.read_info to prevent the unwaranted warning.

### CODE REVIEW

**Review by m-richards — CHANGES_REQUESTED** [review]
Thanks for reporting this and providing a fix @FredericF! Would you be able to add a regression test to 
geopandas/io/tests/test_file.py (say taking your example from the issue) to ensure we make sure this keeps working in the future?

You can put the `read_file` call inside a 
```python
with warnings.catch_warnings():
    warnings.simplefilter("error")
```
block to make sure no warnings are shown.

**Comment by FredericF** [comment]
Hello @m-richards, I added a test.

**Review by m-richards — APPROVED** [review]
Thanks @FredericF for this, looking good now!

I've just pushed a couple of commits to make this test explicit on the engines being tested given we see this specifically with pyogrio (test is run against fiona too where installed, but that's fine) and a changelog entry - normally I'd leave that to you, but there's no heading to place under given we've just done a release.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 Bug fixes:
 
+- Fix unspecified layer warning being emitted reading multilayer datasets, even
```
Comment (martinfleis) [inline CHANGELOG.md:7]:
```suggestion
- Fix unspecified layer warning being emitted while reading multilayer datasets, even
```

**Review by martinfleis — APPROVED** [review]
One minor nit on the changelog. Thanks!

**Pull request merged**